### PR TITLE
Add similar post endpoint

### PR DESCRIPTION
**Pull request by akmatchev**
## Overview

Added a similar post endpoint that returns a list of posts similar to a post given its ID.

## Changes Made

Added an endpoint that takes in a post ID as a parameter and returns a list of posts that are related to that post.

## Test Coverage

I tested the feature through Postman. Successfully related two posts titled "iPhone" and "iPad". 

## Next Steps (delete if not applicable)

Add unit test cases and possibly implement a tree structure or a sorting method to sort posts in descending relevancy order.

### CODE REVIEW

File: src/api/controllers/UserReviewController.ts
Comment (snajima) [inline src/api/controllers/UserReviewController.ts:1]: these files belong to a separate PR. when you want to work on two different branches, you need to make sure you're on `main` before creating the new branch so commits for a different feature don't get copied over! i'll allow it this time but be careful in the future!

File: src/services/PostService.ts [hunk 1]
```diff
@@ -223,5 +223,33 @@ export class PostService {
...
+  public async similarPosts(params: UuidParam): Promise<PostModel[]> {
+    return this.transactions.readOnly(async (transactionalEntityManager) => {
+      const postRepository = Repositories.post(transactionalEntityManager);
+      const post = await postRepository.getPostById(params.id);
+      if (!post) throw new NotFoundError('Post not found!');
+      const allPosts = await postRepository.getAllPosts();
+      let posts: PostModel[] = []
+      for (const p of allPosts) {
+        if (post.id != p.id) {
+          const model = await getLoadedModel();
```
Comment (snajima) [inline src/services/PostService.ts:236]: can we have this line outside the `for (const p of allPosts) {` for loop? we only need to get the loaded model once per function call

**Review by snajima — APPROVED** [review]
lgtm! i see you've implemented not including the current posts in a pretty easy way c:

**Review by mateow99 — COMMENTED** [review]
LGTM

**Review by mateow99 — APPROVED** [review]
LGTM

File: src/api/controllers/UserReviewController.ts
Comment (akmatchev) [reply src/api/controllers/UserReviewController.ts:1]: Sorry about that, won't happen in the future.

**Pull request merged**